plugins/bazaar/src/components/EntityBazaarInfoCard/EntityBazaarInfoCard.tsx

```diff
@@ -0,0 +1,345 @@
+import PersonAddIcon from '@material-ui/icons/PersonAdd';
+import MoreVertIcon from '@material-ui/icons/MoreVert';
+import DeleteIcon from '@material-ui/icons/Delete';
+import { EditProjectDialog } from '../EditProjectDialog';
+import { DeleteProjectDialog } from '../DeleteProjectDialog';
+import ExitToAppIcon from '@material-ui/icons/ExitToApp';
+import { useApi, identityApiRef } from '@backstage/core-plugin-api';
+import { useAsync } from 'react-use';
+import { Member, BazaarProject } from '../../types';
+import { bazaarApiRef } from '../../api';
+import { stringifyEntityRef } from '@backstage/catalog-model';
+
+const useStyles = makeStyles({
+  description: {
+    wordBreak: 'break-word',
+  },
+  icon: {
+    marginRight: '1.75rem',
+  },
+  link: {
+    color: '#9cc9ff',
+    '&:hover': {
+      textDecoration: 'underline',
+    },
+  },
+  memberLink: {
+    display: 'block',
+    marginBottom: '0.3rem',
+  },
+});
+
+const sortMembers = (m1: Member, m2: Member) => {
+  return new Date(m2.joinDate).getTime() - new Date(m1.joinDate).getTime();
+};
+
+export const EntityBazaarInfoCard = () => {
+  const { entity } = useEntity();
+  const classes = useStyles();
+  const bazaarApi = useApi(bazaarApiRef);
+  const identity = useApi(identityApiRef);
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement>();
+  const [open, setOpen] = useState(false);
+  const [popoverOpen, setPopoverOpen] = useState(false);
+  const [openDelete, setOpenDelete] = useState(false);
+
+  const [isMember, setIsMember] = useState(false);
+  const [members, setMembers] = useState<Member[]>([]);
+  const [bazaarProject, setBazaarProject] = useState<BazaarProject>({
+    entityRef: '',
+    name: '',
+    community: '',
+    announcement: '',
+    status: 'proposed',
+    updatedAt: '',
+  });
+  const [isBazaar, setIsBazaar] = useState(false);
+
+  const getInitMemberStatus = async () => {
+    const response = await bazaarApi.getMembers(entity);
+    const dbMembers = response.data.map((obj: any) => {
+      const member: Member = {
+        userId: obj.user_id,
+        entityRef: obj.entity_ref,
+        joinDate: obj.join_date,
+      };
+
+      return member;
+    });
+
+    dbMembers.sort(sortMembers);
+
+    setMembers(dbMembers);
+    setIsMember(
+      dbMembers
+        .map((member: Member) => member.userId)
+        .indexOf(identity.getUserId()) >= 0,
+    );
+  };
+
+  const getMetadata = async () => {
+    const response = await bazaarApi.getMetadata(entity);
+
+    if (response.status !== 404) {
+      setIsBazaar(true);
+      const data = await response.json().then((resp: any) => resp.data);
+
+      setBazaarProject({
+        entityRef: data[0].entityRef,
+        name: data[0].name,
+        community: data[0].community,
+        announcement: data[0].announcement,
+        status: data[0].status,
+        updatedAt: data[0].updatedAt,
+      });
+    }
+  };
+
+  const { loading } = useAsync(async () => {
```

Comment:
the canonical way of doing this loading is rather to do `const { loading, error, value } = useAsync(...` and then make the function _return_ the data it fetched as a structured object inside `value`. No need to redundantly set a useState thing as well on top of that from inside this thing.

No need to fix urgently, but had to note it